pandas/_libs/missing.pyx

```diff
@@ -248,17 +251,11 @@ def isnaobj2d_old(arr: ndarray) -> ndarray:
 
 
 def isposinf_scalar(val: object) -> bool:
-    if util.is_float_object(val) and val == INF:
-        return True
-    else:
-        return False
+    return util.is_float_object(val) and val == INF
```

Comment:
i maybe wrong but if is_float_object returns PyFloat_Check or ... and PyFloat_Check is bint, then 0 and True would return 0. or does cython not work like that

Reply:
is the question about correctness or about types?  in this case cython will handle the types correctly so when called from python this will return bool

Reply:
thanks. I thought that maybe the original code was intentional